Mathlib/RingTheory/MvPolynomial/BinaryForm.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.RingTheory.MvPolynomial.Homogeneous
+
+/-!
+# Binary forms
+
+A binary form of degree `d` is a homogeneous polynomial in two variables of degree `d`.
+
+## Main definitions
+
+- `BinaryForm R d` is the type of binary forms over a commutative ring `R` of degree `d`.
+- `BinaryForm.basis R d` is the basis of `BinaryForm R d` consisting of the monomials
+  `X ^ i * Y ^ (d - i)` for `i : Fin (d + 1)`.
+
+-/
+
+
+universe u
+
+theorem Finsupp.degree_eq_sum {ι : Type*} [DecidableEq ι] [Fintype ι]
+    {R : Type u} [AddCommMonoid R] (f : ι →₀ R) :
+    f.degree = ∑ i, f i := by
+  rw [degree, Finset.sum_congr_of_eq_on_inter] <;> simp
+
+@[simp] theorem Finsupp.mapDomain.linearEquiv_symm {α α' M R : Type*} [Semiring R] [AddCommMonoid M]
+    [Module R M] (f : α ≃ α') :
+    (mapDomain.linearEquiv M R f).symm = mapDomain.linearEquiv M R f.symm := rfl
+
+@[simp] theorem Finsupp.mapDomain.linearEquiv_single
+    {α α' M R : Type*} [Semiring R] [AddCommMonoid M]
+    [Module R M] (f : α ≃ α') (i : α) (a : M) :
+    mapDomain.linearEquiv M R f (single i a) = single (f i) a := by
+  simp [mapDomain.linearEquiv]
+
+@[simp] theorem Finsupp.supportedEquivFinsupp_symm_single
+    {α M R : Type*} [Semiring R] [AddCommMonoid M] [Module R M] (s : Set α) (i : s) (a : M) :
+    ((supportedEquivFinsupp (R := R) s).symm (single i a) : α →₀ M) = single ↑i a := by
+  classical rw [supportedEquivFinsupp_symm_apply_coe, extendDomain_single]
+
+
+open MvPolynomial Finsupp
+
+/-- A binary form of degree `d` is a homogeneous polynomial in two variables of degree `d`. -/
+abbrev BinaryForm (R : Type u) [CommSemiring R] (d : ℕ) : Type u :=
+  homogeneousSubmodule (Fin 2) R d
+
+namespace BinaryForm
+
+variable (R : Type u) [CommSemiring R] (d : ℕ)
+
+/-- The bijection from `Fin (d + 1)` to degrees of bivariate monomials with degree `d`.
```

Comment:
```suggestion
/-- The bijection from `Fin (d + 1)` to degrees of bivariate monomials with total degree `d`.
```
Maybe? I was confused at first by the two notions of "degree" in the same line...